### PR TITLE
Refactoring afterData hook

### PR DESCRIPTION
**Pull request by pazdera**
* afterData renamed to afterDataProcessed
* afterSuccess renamed to afterDataProcessed for consistency
* afterDataProcessed and afterDataReceived both receive endpoint as first param
* added afterDataReceived that receives raw API response
* added ResponseCache

Contains breaking changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,9 +78,28 @@ account.addPlugin({
...
     },
-    // Will be called after the data was received bu before it is resolved to the client user
+    onError(endpoint, error) {
+        if (error === 'fetch failed') {
+            /* Silently spoof cached response */
+            endpoint.response = ResponseCache.lookup(endponint),
+            return Promise.resolve(endpoint);
+        } else {
+            /* Log and rethrow */
+            console.log(error);
```
Comment (paulvarache) [inline README.md]: Logged errors are usually not easy to read. Either use console.error or just throw, the browser will display a thrown error.

**Comment by pazdera** [comment]
@paulvarache The response cache is an abstract way of writing into localStorage by passing in the endpoint object and the raw data. You retrieve data again by passing in the endpoint object.

That means that you don't need to create your own with localStorage keys if you want to cache or load a cached response in a plugin. And also that all cached responses will be stored under `cached:...` so that we know what it is.

**Comment by paulvarache** [comment]
@pazdera That makes sense, but is that enabled by default on every fetch fail? This would cause side effects that are not expected. In some cases, I want to receive the actual error

File: README.md (hunk 1 above)
Comment (paulvarache) [reply README.md]: 👍

**Comment by pazdera** [comment]
@paulvarache No, no, it's not enabled at all. You can optionally import the static class and use it if you want. Like I do here:

https://github.com/KanoComputing/kit-app-ui/pull/320/files#diff-34444541a188dd87e04741815706cc17R2

**Comment by paulvarache** [comment]
Perfect, hard to see in the PR. Didn't realise it was in the docs

**Pull request merged**